src/odemis/driver/orsay.py

```diff
@@ -135,31 +140,40 @@ def _connection_monitor(self):
-            if self._device.HttpConnection._HTTPConnection__response is None or \
-                    self._device.MessageConnection.Connection._HTTPConnection__response is None:
-                self.state._set_value(HwError("Connection to Orsay server lost. Trying to reconnect..."),
-                                      force_write=True)
-                self._device.HttpConnection.close()  # close the current connection
-                self._device.MessageConnection.Connection.close()
-                self._device = None  # destroy the current connection object
-                try:
-                    self._device = Connection(self._host)
-                    time.sleep(1)
-                    self.on_connect()
-                    for child in self.children.value:
-                        child.on_connect()
-                    self.state._set_value(model.ST_RUNNING, force_write=True)
-                except Exception as ex:
-                    logging.exception("Trying to reconnect to Orsay server: %s." % ex)
-            else:
-                self.update_VAs()
-                for child in self.children.value:
-                    child.update_VAs()
-            self._stop_connection_monitor.wait(5)
-
-        logging.debug("Orsay server connection check thread closed")
-        self._stop_connection_monitor.clear()
+        try:
+            while not self._stop_connection_monitor.is_set():
+
+                if self._device and (self._device.HttpConnection._HTTPConnection__response is None or
+                                     self._device.MessageConnection.Connection._HTTPConnection__response is None):
+                    self.state._set_value(HwError("Connection to Orsay server lost. Trying to reconnect..."),
+                                          force_write=True)
+                    self._device.HttpConnection.close()  # close the current connection
+                    self._device.MessageConnection.Connection.close()
+                    self._device = None  # destroy the current connection object
+
+                if not self._device:  # if there currently is no connection
+                    try:  # try to reconnect
+                        self._device = Connection(self._host)
+                        time.sleep(1)
+                        self.on_connect()
+                        for child in self.children.value:
+                            child.on_connect()
+                        self.state._set_value(model.ST_RUNNING, force_write=True)
+                    except Exception as ex:
+                        logging.exception("Trying to reconnect to Orsay server: %s." % ex)
+                else:
+                    try:
+                        self.update_VAs()
+                        for child in self.children.value:
+                            child.update_VAs()
+                    except Exception as e:
+                        logging.exception(e)
```

Comment:
`logging.exception()` is a little special, as it has special code to output the exception. So you don't need to also show the exception. You can just write like:
```python
except Exception:
     logging.exception("Failure while updating VAs")
```
Same thing for the catch just above,  l 163, and just below, l 173.